code/game/gamemodes/objective_items.dm

```diff
@@ -139,6 +139,55 @@
 			return 1
 	return 0
 
+//pets
+/datum/objective_item/steal/iandog //[FULP]
+	name = "Ian, the Head of Personnel's pet corgi, alive."
+	targetitem = /obj/item/pet_carrier
+	difficulty = 20
+	excludefromjob = list("Head of Personnel")
+
+/datum/objective_item/steal/iandog/check_special_completion(obj/item/pet_carrier/F)
+	for(var/mob/living/simple_animal/pet/dog/corgi/Ian/D in F)
+		if(D.stat != DEAD)//checks if pet is alive.
+			return 1

```

Comment:
you dont seem to be checking that the objective is anywhere - this proc should return true so long as Ian is alive in a pet carrier somewhere

Reply:
turns out i am stupid and this works just fine. check_special_completion must have some other code i'm not seeing to check if the object is on you